be/src/exec/iceberg/iceberg_delete_builder.cpp

```diff
@@ -39,40 +39,241 @@ static const IcebergColumnMeta k_delete_file_pos{
         .id = INT32_MAX - 102, .col_name = "pos", .type = TPrimitiveType::BIGINT};
 
 Status ParquetPositionDeleteBuilder::build(const std::string& timezone, const std::string& delete_file_path,
-                                           int64_t file_length, std::set<int64_t>* need_skip_rowids) {
-    std::vector<SlotDescriptor*> slot_descriptors{&(IcebergDeleteFileMeta::get_delete_file_path_slot()),
-                                                  &(IcebergDeleteFileMeta::get_delete_file_pos_slot())};
-    auto iter = std::make_unique<IcebergDeleteFileIterator>();
-    RETURN_IF_ERROR(iter->init(_fs, timezone, delete_file_path, file_length, slot_descriptors, true));
-    std::shared_ptr<::arrow::RecordBatch> batch;
+                                           int64_t file_length, std::set<int64_t>* need_skip_rowids,
+                                           const HdfsScannerParams& scanner_params, RuntimeState* state) {
+    std::unique_ptr<RandomAccessFile> raw_file;
+    ASSIGN_OR_RETURN(raw_file, _fs->new_random_access_file(delete_file_path));
+    std::vector slot_descriptors{&(IcebergDeleteFileMeta::get_delete_file_path_slot()),
+                                 &(IcebergDeleteFileMeta::get_delete_file_pos_slot())};
+
+    HdfsScanStats app_scan_stats;
+    HdfsScanStats fs_scan_stats;
+
+    const int64_t file_size = file_length;
+    raw_file->set_size(file_size);
+    const std::string& filename = raw_file->filename();
+
+    std::shared_ptr<io::SeekableInputStream> input_stream = raw_file->stream();
+
+    input_stream = std::make_shared<CountedSeekableInputStream>(input_stream, &fs_scan_stats);
+
+    auto shared_buffered_input_stream =
+            std::make_shared<io::SharedBufferedInputStream>(input_stream, filename, file_size);
+    const io::SharedBufferedInputStream::CoalesceOptions options = {
+            .max_dist_size = config::io_coalesce_read_max_distance_size,
+            .max_buffer_size = config::io_coalesce_read_max_buffer_size};
+    shared_buffered_input_stream->set_coalesce_options(options);
+    if (file_size < config::io_coalesce_read_max_buffer_size) {
```

Comment:
for (int64_t offset = start_offset; offset < end_offset;) {
        const int64_t remain_length =
                std::min(static_cast<int64_t>(config::io_coalesce_read_max_buffer_size), end_offset - offset);
        disk_ranges.emplace_back(offset, remain_length);
        offset += remain_length;
    }